src/components/MainContent/Components.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { Element } from 'react-scroll/modules';
+
+export interface ContentProps {
+  children:
+    | void
+    | string
+    | JSX.Element
+    | JSX.Element[]
+    | (JSX.Element | undefined)[]
+    | (JSX.Element | undefined);
+}
+
+export interface SectionProps extends ContentProps {
+  name: string;
+}
+
+export const Section = ({ name, children }: SectionProps) => (
+  <Element name={name}>
+    {children}
+    <style jsx>{`
+      padding-left: 0.5em;
+    `}</style>
+  </Element>
+);
+
+export const MainHeader = ({ children }: ContentProps) => (
+  <header>
+    <h1>{children}</h1>
```

Comment:
Is there a situation in which we would put non-string `children` into any sort of header component? I think it might be worth your time to make a new interface that enforces passing string children.